python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -961,6 +962,139 @@ async def close_dome(self, force: bool = True) -> None:
                 f"{ATDome.ShutterDoorState.CLOSED}"
             )
 
+    async def open_dropout_door(self) -> None:
+        """Open ATdome dropout door."""
+
+        dropout_door_state = await self.rem.atdome.evt_dropoutDoorState.aget(
+            timeout=self.fast_timeout
+        )
+
+        if dropout_door_state.state in {
+            ATDome.ShutterDoorState.CLOSED,
+            ATDome.ShutterDoorState.PARTIALLYOPENED,
+        }:
+            self.log.debug("Opening dome dropout door...")
+
+            # flush state
+            self.rem.atdome.evt_dropoutDoorState.flush()
+
+            open_dropout_door_task = asyncio.create_task(
+                self.rem.atdome.cmd_moveShutterDropoutDoor.set_start(
+                    open=True, timeout=self.open_dropout_door_time
+                )
+            )
+
+            self.rem.atdome.evt_summaryState.flush()
+            task_list = [
+                asyncio.create_task(self.check_component_state("atdome")),
+                asyncio.create_task(
+                    self._wait_for_dropout_door_state(
+                        state=ATDome.ShutterDoorState.OPENED,
+                        cmd_task=open_dropout_door_task,
+                        timeout=self.open_dropout_door_time,
+                    )
+                ),
+            ]
+
+            await self.process_as_completed(task_list)
+
+        elif dropout_door_state.state == ATDome.ShutterDoorState.OPENED:
+            self.log.info("ATDome Dropout Door is already opened. Ignoring.")
+        else:
+            raise RuntimeError(
+                f"Dropout Door state is "
+                f"{ATDome.ShutterDoorState(dropout_door_state.state)!r}. "
+                f"Expected either {ATDome.ShutterDoorState.CLOSED!r}, "
+                f"{ATDome.ShutterDoorState.PARTIALLYOPENED!r} or"
+                f"{ATDome.ShutterDoorState.OPENED}"
+            )
+
+    async def close_dropout_door(self) -> None:
+        """Close ATDome dropout door"""
+
+        dropout_door_state = await self.rem.atdome.evt_dropoutDoorState.aget(
+            timeout=self.fast_timeout
+        )
+        if dropout_door_state.state == ATDome.ShutterDoorState.OPENED:
+            self.log.debug("Closing dome dropout door...")
+            await self.rem.atdome.cmd_moveShutterDropoutDoor.set_start(
+                open=False, timeout=self.open_dropout_door_time
+            )
```

Comment:
Need to wait for the dropout to close right?

Reply:
Alright. Take a look to see if it is ok now.